### ISSUE TITLE
Multiple proposals!

### ISSUE THREAD

**Issue by hanyuone** [LogicodeLang/Logicode#25]
_There are ~~three~~ four proposals here._

Proposal 1: Multiple statements in `cond`s.

```
var asdf = 1
cond asdf -> [
    var abcd = 1
    var other = 110
]/[
    var abcd = 0
]
out abcd

Output: 1
```

---

Proposal 2: `@` working for multiple chars at once, something like this:

```
out @(10001,100111,1111010)
```

(This can be extended to all prefix built-in commands).

---

Proposal 3: Null values for variables.

```
var x
# x is now equal to an empty binary string (or represented as [] in processing code)
```

---

Proposal 4: Boolean command.

Uses the `?` symbol (prefix).

```
out ?10011

Output: 1
```

**Comment by bortkiewicz** [comment]
Note on Proposal 3: can't we change it to `var x = void`?

Also, could there be comments, like `var x=1 ?assign x?`

and

```
?This program
prints out 1.?
out 1
```

This is useful since I find Logicode as knotted and twisted as Brainfuck...

**Comment by hanyuone** [comment]
There's already comments, using `#`. So, like this:

```
# This is a line comment
```

As for your tip on Proposal 3, sure, why not. I personally favour my own proposal, though.

**Comment by somebody1234** [comment]
IMO proposal 3 should be `var foo` just makes it ininitialized (i.e. empty)

**Comment by hanyuone** [comment]
@somebody1234 Oh yeah, that could work.

**Comment by bortkiewicz** [comment]
> There's already comments, using #. So, like this:

it's not in the wiki, how do I know??

var foo and var foo = void and var foo = 0 all make sense, however var foo = 0 means something quite different..

**Comment by hanyuone** [comment]
@testitem Yeah, I'm still busy making the docs from scratch... sorry if it's under-documented.

**Comment by bortkiewicz** [comment]
oh. wanna me to help you?

P.S. do we need this?? https://testitemqlstudio.visualstudio.com/Logicode

**Comment by hanyuone** [comment]
@testitem What is that?

**Comment by bortkiewicz** [comment]
Well just a branched version of Logicode... I work on that most of the time and I get ideas from the Github repo's code

**Comment by bortkiewicz** [comment]
So Proposal Number 3 is done, and as "var foo = "?

**Comment by hanyuone** [comment]
@testitem That's changed to `var foo` (so it's done). Also, Proposal 1 is done, as well as Proposal 4. Which just leaves Proposal 2, which I'll get around to later.

**Comment by bortkiewicz** [comment]
Wait what is ? again?

**Comment by hanyuone** [comment]
`?` is actually now a random bit... I forgot about that, so I have to change the boolean thingo to `*`.

**Comment by bortkiewicz** [comment]
No what does it DO?

**Comment by hanyuone** [comment]
`?` generates a random bit, so `0` or `1`.

e.g.

```
var x = ?
out x

Output: 0 or 1
```

`*` is a boolean (i.e. it generates `1` if the variable is non-zero, else it generates `0`)

e.g.

```
var x = *100
out x

Output: 1
```

---

```
var x = *000
out x

Output: 0
```

**Comment by bortkiewicz** [comment]
Oh Thanks.
